### PR TITLE
Read appconfig flag to control admission scenarios (EXPOSUREAPP-11801)

### PR DESCRIPTION
**Pull request by LukasLechnerDev**
The new admission scenario tile is now enabled / disabled depending on the `dcc-admission-check-scenarios-disabled` appconfig flag.

Details: https://github.com/corona-warn-app/cwa-app-tech-spec/pull/157/commits/c0cb645ae4dfddc2a6e4e441248b7e6cd397b729

I also did some refactorings in `CWAConfigMapper` its associated test.

Flag can be enabled/disabled via Mock-Server:

grafik

in text, thanks to Safari:
```js 
cwa app-config set ccl-admission-check-scenarios-disabled 0 --android -e cf
cwa app-config set ccl-admission-check-scenarios-disabled 1 --android -e cf
```

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapper.kt [hunk 1]
```diff
@@ -33,34 +34,15 @@ class CWAConfigMapper @Inject constructor() : CWAConfig.Mapper {
...
-            (0 until appFeatures.appFeaturesCount)
-                .map { appFeatures.getAppFeatures(it) }
-                .firstOrNull { it.label == "unencrypted-checkins-enabled" }
-                ?.let { it.value == 1 }
-                ?: false
-        } catch (e: Exception) {
-            Timber.e(e, "Failed to map `unencrypted-checkins-enabled` from %s", this)
-            false
-        }
-    }
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapper.kt:63]: Created the new extension function `findBoolean()` to avoid duplicated code like this.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapper.kt [hunk 2]
```diff
@@ -103,6 +85,11 @@ class CWAConfigMapper @Inject constructor() : CWAConfig.Mapper {
         }
     }
 
+    private fun ApplicationConfigurationAndroid.dccAdmissionCheckScenariosDisabled() = findBoolean(
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapper.kt:88]: Use the new extension function `findBoolean()` for the new flag.

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapperTest.kt [hunk 3]
```diff
@@ -62,69 +62,96 @@ class CWAConfigMapperTest : BaseTest() {
...
-                        }.build()
-                    )
-                }
-            )
-            .build()
-        createInstance().map(rawConfig).apply {
-            isDeviceTimeCheckEnabled shouldBe false
-            isUnencryptedCheckInsEnabled shouldBe false
-        }
+    fun `findBoolean() should return true on 1`() {
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapperTest.kt:65]: Added some tests for `findBoolean()` which also cover the edge-cases when the flag contains an invalid value or an exception is thrown. These cases are already covered, so we don't need tests for these edge cases for every flag that uses `findBoolean()`.

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapperTest.kt [hunk 4]
```diff
@@ -237,34 +264,7 @@ class CWAConfigMapperTest : BaseTest() {
     }
 
     @Test
-    fun `feature, unencrypted checkins disabled`() {
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapperTest.kt:240]: I removed some test cases like this because they were covered by other tests above already.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapper.kt [hunk 5]
```diff
@@ -103,6 +84,31 @@ class CWAConfigMapper @Inject constructor() : CWAConfig.Mapper {
...
+        return try {
+            val value = (0 until appFeatures.appFeaturesCount)
+                .map { appFeatures.getAppFeatures(it) }
+                .firstOrNull { it.label == labelValue }
+                ?.value
+
+            return when (value) {
+                null -> defaultValue // no item found
+                1 -> true // found item has value '1'
+                else -> defaultValue // found item has value different to '1'
```
[the remark this answers is not in the document]
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapper.kt:104]: The code that you posted has a problem that it returns `false` for every `it.value` different to `1` but it should return `defaultValue` in this case.

This was a very small bug that we had, but it wasn't a problem, since every default value that we have is `false`.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapper.kt [hunk 6]
```diff
@@ -103,6 +84,31 @@ class CWAConfigMapper @Inject constructor() : CWAConfig.Mapper {
         }
     }
 
+    private fun ApplicationConfigurationAndroid.dccAdmissionCheckScenariosDisabled() = findBoolean(
+        labelValue = "dcc-admission-check-scenarios-disabled",
+        defaultValue = DCC_ADMISSION_CHECK_SCENARIOS_DISABLED
```
[the remark this answers is not in the document]
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapper.kt:89]: Removed it completely, see other comment.

**Comment by mtwalli** [comment]
@LukasLechnerDev  did you push your latest changes ? I don't see them here yet

**Comment by LukasLechnerDev** [comment]
@mtwalli Ah, you are right. I forgot to push the "Add default value" commit. Its now pushed.

**Review by mtwalli — APPROVED** [review]
Thanks, LGTM

**Review by chiljamgossow — APPROVED** [review]
Looks good, thanks for refactoring 👍
